addons/mail/models/mail_mail.py

```diff
@@ -78,15 +78,18 @@ def default_get(self, fields):
     auto_delete = fields.Boolean(
         'Auto Delete',
         help="This option permanently removes any track of email after it's been sent, including from the Technical menu in the Settings, in order to preserve storage space of your Odoo database.")
-    scheduled_date = fields.Char('Scheduled Send Date',
+    scheduled_date = fields.Datetime('Scheduled Send Date',
         help="If set, the queue manager will send the email after the date. If not set, the email will be send as soon as possible.")
 
     @api.model_create_multi
     def create(self, values_list):
-        # notification field: if not set, set if mail comes from an existing mail.message
         for values in values_list:
+            # notification field: if not set, set if mail comes from an existing mail.message
             if 'is_notification' not in values and values.get('mail_message_id'):
                 values['is_notification'] = True
+            # scheduled_date field: if not set, set its value to None
```

Comment:
Why would we do that ? Default False value is fine to me, isn't it ?

Reply:
@tde-banana-odoo in case the value was not set in `email_template` the field will hold `''` as a value and will throw a casting issue in (char vs datetime) `mail_mail` when the create method is triggered.
Thanks for the quick review (as usual), I'll autosquash.

Reply:
Can't we write '' in a datetime field ? I did not know that. In that case I would give False instead of None (False being a better default value), and add a small comment as it may be removed (other people may think this code is not necessary).

Reply:
@tde-banana-odoo I previously added `default=False` in the field but it didn't work so I fixed it in the create method where the issue happens. I'll test with `False` instead of `None` and if it goes fine I'll keep it.